unlock_indicator.c

```diff
@@ -86,10 +100,9 @@ auth_state_t auth_state;
  * resolution and returns it.
  *
  */
-xcb_pixmap_t draw_image(uint32_t *resolution) {
+
+xcb_pixmap_t draw_image_with_sf_and_diam(uint32_t *resolution, const double scaling_factor, int button_diameter_physical) {
```

Comment:
I don't quite understand why this function was split like this since you call it with exactly the same sf and button diameter in both places?

Reply:
In `redraw_screen()` I created `scaling_factor` and `button_diameter_physical` variables to calculate actual area of lock to clear (which is passed to `xcb_clear_area()`). `draw_image` function computes `scaling_factor` and `button_diameter_physical` too. So I created new function which takes `scaling_factor` and `button_diameter_physical` as arguments so program wouldn't compute them twice in `redraw_screen`.

It is a small bit of optimization, so I think it can be removed to keep the code more *easy to understand*

Reply:
AFAIK the DPI is always constant, therefore is the scaling factor and thus so is the button diameter. So maybe we should just make those global values?

Reply:
> AFAIK the DPI is always constant, therefore is the scaling factor and thus so is the button diameter. So maybe we should just make those global values?

Yes, `dpi` is constant, `BUTTON_DIAMETER` is defined, but `button_diameter_physical` is neither defined nor constant. We can make them global values but it is too complex to save one computation. Even this split was too much for a few operations

Reply:
> but button_diameter_physical is neither defined nor constant

But it's just a simple arithmetic operation between two constants, so it's effectively constant. But maybe for now let's undo splitting the function and then we can at least move ahead with this PR's main focus. :-)